src/iomei.cpp

```diff
@@ -2718,6 +2718,34 @@ bool MeiInput::ReadDoc(pugi::xml_node root)
         m_doc->ConvertScoreDefMarkupDoc();
     }
 
+    // Upon MEI import: use expansion ID, given by command line argument
+    std::string expansionId = m_doc->GetOptions()->m_useExpansion.GetValue();
+    if (!expansionId.empty()) {
+        Expansion *start = dynamic_cast<Expansion *>(m_doc->FindChildByUuid(expansionId));
+        if (start == NULL) {
+            LogMessage("Import MEI: expansion ID \"%s\" not found.", expansionId.c_str());
+        }
+        else {
+            xsdAnyURI_List expansionList = start->GetPlist();
+            // std::cout << "[useExpansion] xml:id=\"" << expansionId.c_str() << "\" plist={";
+            // for (std::string s : expansionList) std::cout << s.c_str() << ((s != expansionList.back()) ? " " :
+            // "}.\n");
+            xsdAnyURI_List existingList;
+            existingList = m_doc->useExpansion(expansionList, existingList, start);
+            // save original/notated expansion as element in expanded MEI
+            Expansion *originalExpansion = new Expansion();
+            originalExpansion->SetUuid("expansion-notated");
```

Comment:
I don't really like this id... Maybe we need to add a bool member to flags them as expansion? See m_isAttribute example

Reply:
Do I understand correctly that the `m_isAttribute` flag is used to output the MEI content that could be written both as elements or attributes is output as it was at input?

However, for an `isExpansion` flag, there is no valid MEI to reflect this in a MEI file. That's why I thought it would be helpful to have one expansion element written that remembers the notated section configuration.

This flag would be useful for displaying repetitions dynamically in an app, so it could be used in parallel with the added expansion element?

Reply:
`m_isAttribute` is used to flag an Object when it is converted from an MEI attribute into an element by verovio. For example `note@accid` become `note/accid@accid` in verovio, so we can support both but implement only the most complex one. When we output the MEI, we can convert it back to an attribute. Potentially, we could ignore it in the output and use verovio to convert some attributes to elements.

It looks like `m_isExpansion` would be useful for similar reason. Once flags, the element could be easier not be written in the output, or written, and it that case verovio could be used to expand MEI files.

Reply:
Dear Laurent, I have implemented all requested changes (std::map, --expand, Object::isExpansion, ExpansionMap as separate class), except the functor approach to the recursive UpdateIds(Object*) method. I wont find the time at the moment, but hope to do this soon. Best, Werner